Nodejs/Product/Nodejs/Project/NodejsGeneralPropertyPage.cs

```diff
@@ -38,49 +46,42 @@ internal override CommonProjectNode Project
+            this.control.IsDirty = false;
         }
 
         public override void LoadSettings()
         {
-            this.Loading = true;
-            try
-            {
-                this._control.NodeExeArguments = this.Project.GetUnevaluatedProperty(NodeProjectProperty.NodeExeArguments);
-                this._control.NodeExePath = this.Project.GetUnevaluatedProperty(NodeProjectProperty.NodeExePath);
-                this._control.ScriptFile = this.Project.GetUnevaluatedProperty(CommonConstants.StartupFile);
-                this._control.ScriptArguments = this.Project.GetUnevaluatedProperty(NodeProjectProperty.ScriptArguments);
-                this._control.WorkingDirectory = this.Project.GetUnevaluatedProperty(CommonConstants.WorkingDirectory);
-                this._control.LaunchUrl = this.Project.GetUnevaluatedProperty(NodeProjectProperty.LaunchUrl);
-                this._control.NodejsPort = this.Project.GetUnevaluatedProperty(NodeProjectProperty.NodejsPort);
-                this._control.DebuggerPort = this.Project.GetUnevaluatedProperty(NodeProjectProperty.DebuggerPort);
-                this._control.Environment = this.Project.GetUnevaluatedProperty(NodeProjectProperty.Environment);
+            this.control.NodeExeArguments = this.Project.GetUnevaluatedProperty(NodeProjectProperty.NodeExeArguments);
+            this.control.NodeExePath = this.Project.GetUnevaluatedProperty(NodeProjectProperty.NodeExePath);
+            this.control.ScriptFile = this.Project.GetUnevaluatedProperty(CommonConstants.StartupFile);
+            this.control.ScriptArguments = this.Project.GetUnevaluatedProperty(NodeProjectProperty.ScriptArguments);
+            this.control.WorkingDirectory = this.Project.GetUnevaluatedProperty(CommonConstants.WorkingDirectory);
+            this.control.LaunchUrl = this.Project.GetUnevaluatedProperty(NodeProjectProperty.LaunchUrl);
+            this.control.NodejsPort = this.Project.GetUnevaluatedProperty(NodeProjectProperty.NodejsPort);
+            this.control.DebuggerPort = this.Project.GetUnevaluatedProperty(NodeProjectProperty.DebuggerPort);
+            this.control.Environment = this.Project.GetUnevaluatedProperty(NodeProjectProperty.Environment);
 
-                // Attempt to parse the boolean.  If we fail, assume it is true.
-                if (!Boolean.TryParse(this.Project.GetUnevaluatedProperty(NodeProjectProperty.StartWebBrowser), out var startWebBrowser))
-                {
-                    startWebBrowser = true;
-                }
-                this._control.StartWebBrowser = startWebBrowser;
-            }
-            finally
+            // Attempt to parse the boolean.  If we fail, assume it is true.
+            if (!bool.TryParse(this.Project.GetUnevaluatedProperty(NodeProjectProperty.StartWebBrowser), out var startWebBrowser))
             {
-                this.Loading = false;
+                startWebBrowser = true;
```

Comment:
Should probably fix the other bug I discussed with you before making this change, where we spin up a thread that waits forever for the Node.js web port to be listening before it opens the browser.